### ISSUE TITLE
MQTTSubscribe value error: start time is greater than stop time

### ISSUE THREAD

**Issue by pmendiuk** [bellrichm/WeeWX-MQTTSubscribe#23]
I installed MQTTSubscribe a couple of days ago.  It is running in weewx 3.9.2 with weatherflowudp and mqtt.   Weewx has shut down a few times today with this type of error:

```
Jun 20 09:32:42 MITX-6930 weewx[14061]: MQTTSubscribeService: Packet after update is: 2019-06-20 09:32:39 PDT (1561048359) dateTime: 1561048359, usUnits: 17, windDir: 0, windSpeed: 0.0
Jun 20 09:32:42 MITX-6930 weewx[14061]: restx: MQTT: Published record 2019-06-20 09:32:39 PDT (1561048359)
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Packet prior to update is: 2019-06-20 09:32:42 PDT (1561048362) dateTime: 1561048362, usUnits: 17, windDir: 0, windSpeed: 0.0
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Processing interval: 1561048359.000000 1561048362.000000
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribe: TopicManager queue size is: 0
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Queue was empty
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Packet after update is: 2019-06-20 09:32:42 PDT (1561048362) dateTime: 1561048362, usUnits: 17, windDir: 0, windSpeed: 0.0
Jun 20 09:32:45 MITX-6930 weewx[14061]: restx: MQTT: Published record 2019-06-20 09:32:42 PDT (1561048362)
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Packet prior to update is: 2019-06-20 09:32:17 PDT (1561048337) dateTime: 1561048337, illuminance: 40134, precipitationType: 0, radiation: 334, rain: 0.0, skyBatteryVoltage: 3.37, usUnits: 17, UV: 3.77, windBatteryStatus: 3.37, windLull: 0.0
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Processing interval: 1561048362.000000 1561048337.000000
Jun 20 09:32:45 MITX-6930 weewx[14061]: engine: Main loop exiting. Shutting engine down.
Jun 20 09:32:45 MITX-6930 weewx[14061]: engine: Shutting down StdReport thread
Jun 20 09:32:45 MITX-6930 weewx[14061]: engine: StdReport thread has been terminated
Jun 20 09:32:45 MITX-6930 weewx[14061]: restx: Shut down MQTT thread.
Jun 20 09:32:45 MITX-6930 weewx[14061]: restx: Shut down Wunderground-RF thread.
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribe: Disconnected with result code 0
Jun 20 09:32:45 MITX-6930 weewx[14061]: engine: Caught unrecoverable exception in engine:
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****  start time (1561048362) is greater than stop time (1561048337)
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****  Traceback (most recent call last):
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/weewx/engine.py", line 890, in main
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      engine.run()
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/weewx/engine.py", line 191, in run
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      self.dispatchEvent(weewx.Event(weewx.NEW_LOOP_PACKET, packet=packet))
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/weewx/engine.py", line 224, in dispatchEvent
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      callback(event)
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/user/MQTTSubscribe.py", line 609, in new_loop_packet
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      target_data = self.subscriber.get_accumulated_data(topic, start_ts, self.end_ts, event.packet['usUnits'])
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/user/MQTTSubscribe.py", line 529, in get_accumulated_data
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      return self.manager.get_accumulated_data(topic, start_ts, end_ts, units)
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/user/MQTTSubscribe.py", line 277, in get_accumulated_data
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      accumulator = weewx.accum.Accum(weeutil.weeutil.TimeSpan(start_ts, end_ts))
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/weeutil/weeutil.py", line 256, in __new__
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      raise ValueError("start time (%d) is greater than stop time (%d)" % (args[0], args[1]))
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****  ValueError: start time (1561048362) is greater than stop time (1561048337)
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****  Exiting.
```
I have added more debugging to syslog and hope to get more info.  Previous to installing MQTTSubscribe, weewx was accepting that data without issue from weatherflowudp.

**Comment by pmendiuk** [comment]
Here is another error event:
```
Jun 20 09:32:42 MITX-6930 weewx[14061]: MQTTSubscribeService: Packet after update is: 2019-06-20 09:32:39 PDT (1561048359) dateTime: 1561048359, usUnits: 17, windDir: 0, windSpeed: 0.0
Jun 20 09:32:42 MITX-6930 weewx[14061]: restx: MQTT: Published record 2019-06-20 09:32:39 PDT (1561048359)
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Packet prior to update is: 2019-06-20 09:32:42 PDT (1561048362) dateTime: 1561048362, usUnits: 17, windDir: 0, windSpeed: 0.0
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Processing interval: 1561048359.000000 1561048362.000000
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribe: TopicManager queue size is: 0
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Queue was empty
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Packet after update is: 2019-06-20 09:32:42 PDT (1561048362) dateTime: 1561048362, usUnits: 17, windDir: 0, windSpeed: 0.0
Jun 20 09:32:45 MITX-6930 weewx[14061]: restx: MQTT: Published record 2019-06-20 09:32:42 PDT (1561048362)
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Packet prior to update is: 2019-06-20 09:32:17 PDT (1561048337) dateTime: 1561048337, illuminance: 40134, precipitationType: 0, radiation: 334, rain: 0.0, skyBatteryVoltage: 3.37, usUnits: 17, UV: 3.77, windBatteryStatus: 3.37, windLull: 0.0
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribeService: Processing interval: 1561048362.000000 1561048337.000000
Jun 20 09:32:45 MITX-6930 weewx[14061]: engine: Main loop exiting. Shutting engine down.
Jun 20 09:32:45 MITX-6930 weewx[14061]: engine: Shutting down StdReport thread
Jun 20 09:32:45 MITX-6930 weewx[14061]: engine: StdReport thread has been terminated
Jun 20 09:32:45 MITX-6930 weewx[14061]: restx: Shut down MQTT thread.
Jun 20 09:32:45 MITX-6930 weewx[14061]: restx: Shut down Wunderground-RF thread.
Jun 20 09:32:45 MITX-6930 weewx[14061]: MQTTSubscribe: Disconnected with result code 0
Jun 20 09:32:45 MITX-6930 weewx[14061]: engine: Caught unrecoverable exception in engine:
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****  start time (1561048362) is greater than stop time (1561048337)
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****  Traceback (most recent call last):
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/weewx/engine.py", line 890, in main
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      engine.run()
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/weewx/engine.py", line 191, in run
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      self.dispatchEvent(weewx.Event(weewx.NEW_LOOP_PACKET, packet=packet))
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/weewx/engine.py", line 224, in dispatchEvent
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      callback(event)
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/user/MQTTSubscribe.py", line 609, in new_loop_packet
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      target_data = self.subscriber.get_accumulated_data(topic, start_ts, self.end_ts, event.packet['usUnits'])
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/user/MQTTSubscribe.py", line 529, in get_accumulated_data
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      return self.manager.get_accumulated_data(topic, start_ts, end_ts, units)
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/user/MQTTSubscribe.py", line 277, in get_accumulated_data
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      accumulator = weewx.accum.Accum(weeutil.weeutil.TimeSpan(start_ts, end_ts))
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****    File "/usr/share/weewx/weeutil/weeutil.py", line 256, in __new__
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****      raise ValueError("start time (%d) is greater than stop time (%d)" % (args[0], args[1]))
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****  ValueError: start time (1561048362) is greater than stop time (1561048337)
Jun 20 09:32:45 MITX-6930 weewx[14061]:     ****  Exiting.
```

**Comment by bellrichm** [comment]
MQTTSubscribe does expect that the packet data comes from the driver in chronological order (I would think other applications would mis-behave in some form too). In order to not crash WeeWX, I will put a check in for packets that have "traveled back in time" and ignore them. When  the next packet with a time that is not in the past arrives, the MQTT queue will be processed and added to that packet.
